File: python/23.05/45.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        ans = [0] + [float('inf')] * len(nums)
        for i in range(len(nums)):
            for j in range(nums[i]):
                if len(ans) > (i + j + 1):
                    ans[i + j + 1] = min(ans[i] + 1, ans[i+j+1])

        return ans[len(nums)-1]

File: typescript/23.08/45.ts
function jump(nums: number[]): number {
    let [end, arr, ans] = [0, 0, 0]
    
    for (let i=0; i<nums.length-1; i++){
        arr = Math.max(arr, nums[i]+i)
        
        if (i === end){
            end = arr
            ans++
        }
    }
    return ans
};
